### DISCUSSION TITLE
Settings Sync btw Mac/Win with Connection Settings

### DISCUSSION THREAD

**Discussion by ScottKlement** [codefori/vscode-ibmi#994] [General]
I love Code's "Settings Sync" because I use 4 different computers, and switch between them often.

Problem is:  Two of them are running MacOS.

Every time I switch to a different OS, it breaks my "Connection Settings" in Code for i because the path names for the SSH keys are different on Mac vs. Windows.

For example, this works fine on MacOS:
```
        {
            "name": "i.scottklement.com",
            "host": "i.scottklement.com",
            "port": 22,
            "username": "sklement",
            "privateKey": "/Users/scott/.ssh/id_rsa"
        }

```
But then when I switch to Windows, I have to change the privateKey setting to this (don't ask why my name is "micro" -- it just is.)
```
        {
            "name": "i.scottklement.com",
            "host": "i.scottklement.com",
            "port": 22,
            "username": "sklement",
            "privateKey": "C:\\Users\\micro\\.ssh\\id_rsa"
        }

```

I've tried using ${userHome} in the privateKey to no avail.  I don't know if it doesn't work there?  Or perhaps it's the different directory name separators that is the problem?

Does anyone have a solution for this?  Or, would it be possible to add a solution somehow into Code for i?  (I'd be willing to make the change if someone can help me get started.)

I could use passwords, of course, but....    yuck.

**Comment by worksofliam** [comment]
I guess the current work around is to have two entries. One for Windows and one for Mac.

We could potentially do something like this, but it would take some looking into. Good idea, though.

```
{
  privateKey: {windows: `C:/id_rsa`, mac: `/Users/scott/.ssh/id_rsa`}
}
```

**Reply by ScottKlement** [reply]
Digging a bit deeper into the docs for Code, I found these variables: https://code.visualstudio.com/docs/editor/variables-reference

After reading it, I expected that I could do something like the following (which sounds perfect):
```
        {
            "name": "i.scottklement.com",
            "host": "i.scottklement.com",
            "port": 22,
            "username": "sklement",
            "privateKey": "${userHome}${pathSeparator}.ssh${pathSeparator}id_rsa"
            //"privateKey": "/Users/micro/.ssh/id_rsa"
        }

```
Unfortunately, I get an error telling me it can't find the private key.   Does Code for i expand these variables?

**Reply by worksofliam** [reply]
It does not, but it's likely we can easily expand them!

If you wanted to take a go at it, we are open to PRs. The change should likely happen in the [`connect` method](https://github.com/halcyon-tech/vscode-ibmi/blob/0045a9cd8608b75f6f03760e7244d700fbda950b/src/api/IBMi.ts#L110) as part of `IBMi`.

* Path separator can be fetched in the `path` module
* Home directory can be fetched in the `os` module

This change should likely be documented too. Likely in the [Connections section](https://github.com/halcyon-tech/docs/blob/main/docs/pages/settings/global.md#connections) in the docs.
